release.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+set -e
+
+function is_version_valid() {
+  echo "$1" | grep -q -E '^[0-9]+\.[0-9]+\.[0-9]+$'
+  if [[ "$?" == "0" ]]; then return 0; else return 1; fi
+}
+
+git reset --hard HEAD > /dev/null 2>&1
```

Comment:
This appears a bit harsh.
It happens to me sometimes that I made some changes that I do not want to lose, and figure it out the moment I release.

Maybe aborting the releasing, and pin pointing the fact that the current state is not clean is better?

Reply:
We could also stash at the beginning and stash pop at the end.

Reply:
I'd rather not stash in scripts, I had bad surprises in the past.

I'd go for either:
1. Raymond' solution where we just abort if the worktree isn't clear
2. Just reset the `transport.go` and `ChangeLog.md` files as those are the only one we modify.

WDYT @rayrutjes @julienbourdeau? 1. looks safer to me indeed.